Tests/UnitTests/EdgeEventErrorTests.swift

```diff
@@ -141,21 +143,27 @@ class EdgeEventErrorTests: XCTestCase {
         let report = EdgeErrorReport(eventIndex: 1, errors: ["error1", "error2"], requestId: "1234", orgId: "abcd")
         let error = EdgeEventError(title: "Test Error", detail: "details", status: 200, type: "error", report: report)
 
-        let encoded = error.asDictionary()
-
-        XCTAssertNotNil(encoded)
-        XCTAssertEqual(5, encoded?.count)
-        XCTAssertEqual("Test Error", encoded?["title"] as? String)
-        XCTAssertEqual("details", encoded?["detail"] as? String)
-        XCTAssertEqual(200, encoded?["status"] as? Int)
-        XCTAssertEqual("error", encoded?["type"] as? String)
-
-        let encodedReport = encoded?["report"] as? [String: Any]
-        XCTAssertNotNil(encodedReport)
-        XCTAssertEqual(3, encodedReport?.count) // eventIndex is no encoded
-        XCTAssertEqual(["error1", "error2"], encodedReport?["errors"] as? [String])
-        XCTAssertEqual("1234", encodedReport?["requestId"] as? String)
-        XCTAssertEqual("abcd", encodedReport?["orgId"] as? String)
+        var encoded = error.asDictionary()
+
+        let expectedJSON = #"""
+        {
+          "detail": "details",
+          "report": {
+            "errors": [
+              "error1",
+              "error2"
+            ],
+            "orgId": "abcd",
+            "requestId": "1234"
+          },
+          "status": 200,
+          "title": "Test Error",
+          "type": "error"
+        }
+        """#
+
+        assertEqual(expected: getAnyCodable(expectedJSON)!,
```

Comment:
This is more of an idea around the assertEqual method. What do you think about overloading the method to take the raw json instead, and internally doing the anycodable conversions? Will make the API easier to use in our tests.